src/intel_pmu.c

```diff
@@ -0,0 +1,512 @@
+    {.name = "L1-icache-prefetches",
+     .config = (PERF_COUNT_HW_CACHE_L1I | HW_CACHE_PREFETCH_ACCESS)},
+    {.name = "L1-icache-prefetch-misses",
+     .config = (PERF_COUNT_HW_CACHE_L1I | HW_CACHE_PREFETCH_MISS)},
+
+    {.name = "LLC-loads",
+     .config = (PERF_COUNT_HW_CACHE_LL | HW_CACHE_READ_ACCESS)},
+    {.name = "LLC-load-misses",
+     .config = (PERF_COUNT_HW_CACHE_LL | HW_CACHE_READ_MISS)},
+    {.name = "LLC-stores",
+     .config = (PERF_COUNT_HW_CACHE_LL | HW_CACHE_WRITE_ACCESS)},
+    {.name = "LLC-store-misses",
+     .config = (PERF_COUNT_HW_CACHE_LL | HW_CACHE_WRITE_MISS)},
+    {.name = "LLC-prefetches",
+     .config = (PERF_COUNT_HW_CACHE_LL | HW_CACHE_PREFETCH_ACCESS)},
+    {.name = "LLC-prefetch-misses",
+     .config = (PERF_COUNT_HW_CACHE_LL | HW_CACHE_PREFETCH_MISS)},
+
+    {.name = "dTLB-loads",
+     .config = (PERF_COUNT_HW_CACHE_DTLB | HW_CACHE_READ_ACCESS)},
+    {.name = "dTLB-load-misses",
+     .config = (PERF_COUNT_HW_CACHE_DTLB | HW_CACHE_READ_MISS)},
+    {.name = "dTLB-stores",
+     .config = (PERF_COUNT_HW_CACHE_DTLB | HW_CACHE_WRITE_ACCESS)},
+    {.name = "dTLB-store-misses",
+     .config = (PERF_COUNT_HW_CACHE_DTLB | HW_CACHE_WRITE_MISS)},
+    {.name = "dTLB-prefetches",
+     .config = (PERF_COUNT_HW_CACHE_DTLB | HW_CACHE_PREFETCH_ACCESS)},
+    {.name = "dTLB-prefetch-misses",
+     .config = (PERF_COUNT_HW_CACHE_DTLB | HW_CACHE_PREFETCH_MISS)},
+
+    {.name = "iTLB-loads",
+     .config = (PERF_COUNT_HW_CACHE_ITLB | HW_CACHE_READ_ACCESS)},
+    {.name = "iTLB-load-misses",
+     .config = (PERF_COUNT_HW_CACHE_ITLB | HW_CACHE_READ_MISS)},
+
+    {.name = "branch-loads",
+     .config = (PERF_COUNT_HW_CACHE_BPU | HW_CACHE_READ_ACCESS)},
+    {.name = "branch-load-misses",
+     .config = (PERF_COUNT_HW_CACHE_BPU | HW_CACHE_READ_MISS)},
+};
+
+event_info_t g_sw_events[] = {
+    {.name = "cpu-clock", .config = PERF_COUNT_SW_CPU_CLOCK},
+
+    {.name = "task-clock", .config = PERF_COUNT_SW_TASK_CLOCK},
+
+    {.name = "context-switches", .config = PERF_COUNT_SW_CONTEXT_SWITCHES},
+
+    {.name = "cpu-migrations", .config = PERF_COUNT_SW_CPU_MIGRATIONS},
+
+    {.name = "page-faults", .config = PERF_COUNT_SW_PAGE_FAULTS},
+
+    {.name = "minor-faults", .config = PERF_COUNT_SW_PAGE_FAULTS_MIN},
+
+    {.name = "major-faults", .config = PERF_COUNT_SW_PAGE_FAULTS_MAJ},
+
+    {.name = "alignment-faults", .config = PERF_COUNT_SW_ALIGNMENT_FAULTS},
+
+    {.name = "emulation-faults", .config = PERF_COUNT_SW_EMULATION_FAULTS},
+};
+
+static intel_pmu_ctx_t g_ctx;
+
+#if COLLECT_DEBUG
+static void pmu_dump_events() {
+
+  DEBUG(PMU_PLUGIN ": Events:");
+
+  struct event *e;
+
+  for (e = g_ctx.event_list->eventlist; e; e = e->next) {
+    DEBUG(PMU_PLUGIN ":   event       : %s", e->event);
+    DEBUG(PMU_PLUGIN ":     group_lead: %d", e->group_leader);
+    DEBUG(PMU_PLUGIN ":     end_group : %d", e->end_group);
+    DEBUG(PMU_PLUGIN ":     type      : 0x%X", e->attr.type);
```

Comment:
Please use the `%#X` format instead of writing out the `0x` manually. Here and below.